### PR TITLE
Update our doc code-gen now that we have synchronous data sources.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
You can see what this looks like here: https://github.com/pulumi/pulumi-aws/pull/696

### CODE REVIEW

File: gen/nodejs/generator.go [hunk 1]
```diff
@@ -651,16 +638,7 @@ func (g *generator) generateResource(r *il.ResourceNode) error {
...
-				// If the input properties did not contain any outputs, then we need to wrap the result in a call to
-				// pulumi.output. Otherwise, we are okay as-is: the apply rewrite perfomed by computeProperty will hav
-				// ensured that the result is output-typed.
-				fmtstr := "%s%s.push(pulumi.output(%s));\n"
-				if transformed {
-					fmtstr = "%s%s.push(%s);\n"
-				}
-
-				g.Printf(fmtstr, g.Indent, name, qualifiedMemberName, inputs)
+				g.Printf("%s%s.push(%s);\n", g.Indent, name, qualifiedMemberName)
```
Comment (CyrusNajmabadi) [inline gen/nodejs/generator.go]: we *were* dropping `inputs` before.   we should fix this up to actually pass these along properly.   i'll see if i can do that.

Comment (pgavlin) [reply gen/nodejs/generator.go]: Ah, because the old code was missing a %s?

**Comment by pgavlin** [comment]
This is mostly OK, but IMO we should only do this for new-enough versions of the SDK. See e.g. `supportsProxyApplies`.

**Comment by pgavlin** [comment]
I also suspect that there is probably some subtle typing issue somewhere w.r.t. data sources: IIRC the binder will type them as `Output`s, so they will trigger the apply transform. We'll need to change that.

File: gen/nodejs/generator.go (hunk 1 above)
Comment (CyrusNajmabadi) [reply gen/nodejs/generator.go]: yes.  but i tried adding it it in, and it's not desirable ether.  Have opened https://github.com/pulumi/tf2pulumi/issues/108 to help track this.

**Comment by CyrusNajmabadi** [comment]
> I also suspect that there is probably some subtle typing issue somewhere w.r.t. data sources: IIRC the binder will type them as Outputs, so they will trigger the apply transform. We'll need to change that.

Agreed.  Are you ok wiht this goign in in the current form?  Would you prefer not to take this?  Do you want to pick this up?  My timeboxing is almost spent :)

**Comment by pgavlin** [comment]
> Are you ok wiht this goign in in the current form?

I would prefer that this not go in until we're sure that we've done the deep work.

File: tests/terraform/aws/asg/variables.tf [hunk 2]
```diff
@@ -12,7 +12,7 @@ variable "aws_amis" {
 }
 
 variable "availability_zones" {
-  default     = "us-east-1b,us-east-1c,us-east-1d,us-east-1e"
```
Comment (pgavlin) [inline tests/terraform/aws/asg/variables.tf]: Did `us-east-1b` disappear?

**Comment by CyrusNajmabadi** [comment]
Ok.  I'll close out.  I don't have the time to get things to that level.

**Pull request closed without merging**